packages/console/src/hooks/use-tenants.ts

```diff
@@ -0,0 +1,86 @@
+import { useLogto } from '@logto/react';
+import { type TenantInfo } from '@logto/schemas';
+import { type Optional, trySafe } from '@silverhand/essentials';
+import { useCallback, useContext, useEffect, useMemo, useState } from 'react';
+import useSWR from 'swr';
+
+import { useCloudApi } from '@/cloud/hooks/use-cloud-api';
+import { TenantsContext } from '@/contexts/TenantsProvider';
+
+import useSwrFetcher from './use-swr-fetcher';
+
+const useTenants = () => {
+  const cloudApi = useCloudApi();
+  const { signIn, getAccessToken } = useLogto();
+  const {
+    tenants,
+    setTenants: originalSetTenants,
```

Comment:
If we use swr, we don't need to keep the tenants in context any more. When you need tenants somewhere else, you can just call useSwr again